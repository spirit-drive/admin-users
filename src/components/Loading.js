import React from 'react';
import PropTypes from 'prop-types';
import { View, StyleSheet, ActivityIndicator } from 'react-native';

const Loading = ({ style }) => (
  <View style={[styles.root, style]}>
    <ActivityIndicator size="small" color="#43D1DE" />
  </View>
);

Loading.propTypes = {
  style: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
};

Loading.defaultProps = {
  style: undefined,
};

const styles = StyleSheet.create({
  root: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
});

export default Loading;
