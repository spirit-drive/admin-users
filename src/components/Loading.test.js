import React from 'react';
import renderer from 'react-test-renderer';
import Loading from './Loading';

it('render Loading', () => {
  expect(renderer.create(<Loading />).toJSON()).toMatchSnapshot();
});
