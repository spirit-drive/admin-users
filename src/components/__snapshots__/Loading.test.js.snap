// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Loading 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#43D1DE"
    hidesWhenStopped={true}
    size="small"
  />
</View>
`;
